ES|QL tab added to Timeline (#4065)
elastic/security-docs: docs/events/timeline-ui-overview.asciidoc

* Specify the date and time range that you want to investigate.
* Reorder the columns and choose which fields to display.
* Choose a data view and whether to show detection alerts only.

[discrete]
[[esql-in-timeline]]

> == Use {esql} to interact with alerts 

Author: Changing "alerts" to "events" or "documents" throughout this section as per @paulewing's request via slack.

```suggestion
== Use {esql} to interact with events 
```

> == Use {esql} to interact with events 

Author: ```suggestion
== Use {esql} to investigate events 
```

preview::[]

> The {ref}/esql.html[Elasticsearch Query Language ({esql})] provides a powerful way to filter, transform, and analyze event data stored in Elasticsearch. {esql} queries use "pipes" to manipulate and transform data in a step-by-step fashion. This approach allows you to compose a series of operations, where the output of one operation becomes the input for the next, enabling complex data transformations and analysis.

Reviewer: ```suggestion
The {ref}/esql.html[Elasticsearch Query Language ({esql})] provides a powerful way to filter, transform, and analyze event data stored in {es}. {esql} queries use "pipes" to manipulate and transform data in a step-by-step fashion. This approach allows you to compose a series of operations, where the output of one operation becomes the input for the next, enabling complex data transformations and analysis.
```

> The {ref}/esql.html[Elasticsearch Query Language ({esql})] provides a powerful way to filter, transform, and analyze alert data stored in Elasticsearch. {esql} queries use "pipes" to manipulate and transform data in a step-by-step fashion. This approach allows you to compose a series of operations, where the output of one operation becomes the input for the next, enabling complex data transformations and analysis.

Author: ```suggestion
The {ref}/esql.html[Elasticsearch Query Language ({esql})] provides a powerful way to filter, transform, and analyze event data stored in Elasticsearch. {esql} queries use "pipes" to manipulate and transform data in a step-by-step fashion. This approach allows you to compose a series of operations, where the output of one operation becomes the input for the next, enabling complex data transformations and analysis.
```

> You can use ES|QL in Timeline by opening the **ES|QL** tab. From the tab, you can: 

Reviewer 2: ```suggestion
You can use ES|QL in Timeline by opening the **ES|QL** tab. From there, you can: 
```

> - Explore your alerts using the default query, or creating a custom one. The default query searches the Security alert index (`.alerts-security.alerts-default`) and  indices specified in the <<update-sec-indices,Security data view>> for alerts that were generated during the last 15 minutes. The query's output is limited to the top ten results.

Reviewer 2: ```suggestion
- Explore your alerts using the default query, or a custom one. The default query searches the Security alert index (`.alerts-security.alerts-default`) and  indices specified in the <<update-sec-indices,Security data view>> for alerts that were generated during the last 15 minutes. The query's output is limited to the top ten results.
```

Reviewer 2: The top ten alerts by what? Severity? Frequency?

Author: I'm not actually sure how the `[LIMIT](https://security-docs_4062.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/esql-commands.html#esql-limit)` command decides which rows/alerts count as the "top" ones. @michaelolo24 do you have any insight on this?

Author: Resolving this as per our discussion in Slack.

> - Explore your alerts using the default query, or create a custom one. The default query searches the Security alert index (`.alerts-security.alerts-default`) and  indices specified in the <<update-sec-indices,Security data view>> for alerts that were generated during the last 15 minutes. The query's output is limited to the top ten results.

Author: ```suggestion
- Explore your events using the default query, or create a custom one. The default query searches documents within the Security alert index (`.alerts-security.alerts-default`) and indices specified in the <<update-sec-indices,Security data view>>, then returns 10 events from the defined time range.  
```

- Click the help icon (image:images/esql-ref-button.png[Click the ES|QL reference button,20,20]) on the far right side of the query editor to open the in-product reference documentation for all ES|QL commands and functions.  

> - Visualize query results using {kibana-ref}/discover.html[Discover] functionality that's available in the tab.

Reviewer 2: ```suggestion
- Visualize query results using {kibana-ref}/discover.html[Discover] functionality.
```

[role="screenshot"]
image::images/esql-tab.png[a Timeline's ES|QL tab]

[discrete]
[[esql-in-timeline-resources]]
=== Additional {esql} resources 

> To get started using {esql}, read the tutorial for {ref}/esql.html[using {esql} in Kibana]. Much of the functionality available in Kibana is also available in Timeline.  

Reviewer 3: ```suggestion
To get started using {esql}, read the tutorial for {ref}/esql-kibana.html[using {esql} in Kibana]. Much of the functionality available in Kibana is also available in Timeline.  
```

> To get started using {esql}, read the tutorial for {ref}/esql-kibana.html[using {esql} in Kibana]. Much of the functionality available in Kibana is also available in Timeline.  

Reviewer: ```suggestion
To get started using {esql}, read the tutorial for {ref}/esql-kibana.html[using {esql} in {kib}]. Much of the functionality available in {kib} is also available in Timeline.  
```

> To get started using ES|QL, read the tutorial for {ref}/esql.html[using ES|QL in Kibana]. Much of the functionality that's offered in Kibana is also offered in Timeline.  

Reviewer 2: ```suggestion
To get started using ES|QL, read the tutorial for {ref}/esql.html[using ES|QL in Kibana]. Much of the functionality available in Kibana is also available in Timeline.  
```
